### ISSUE TITLE
[v8.4.41] Terraform errors related to the overwrite file.

### ISSUE THREAD

**Issue by giva01121** [cloudfoundry/bosh-bootloader#522]
Hello,
we are facing issues while trying to update the version from v8.4.40 to v8.4.41
Currently we have these files:
```

bbl-state/terraform  [bbl_update_v8.4.41]$ ls -la
total 24
drwxr-xr-x   5 givanov  staff   160 Jul  1 17:23 .
drwxr-xr-x  15 givanov  staff   480 Jul  1 17:03 ..
drwxr-xr-x   3 givanov  staff    96 Jul  1 17:23 .terraform
-rwxr-xr-x   1 givanov  staff  7062 Jul  2 11:36 bbl-template.tf
-rw-r--r--   1 givanov  staff  2787 Jul  1 16:53 network_security_rule_override.tf
```
From the terraform documentation it says that we need to name the file with _override:
https://www.terraform.io/docs/language/files/override.html

But still we are getting errors related to non-existing terraform rules which are already in the terraform template. With the older version we didn't face this problem:

```
+ bbl plan --name vault-qa-schwarz --iaas azure
step: generating terraform template
step: generating terraform variables
step: terraform init
There are some problems with the configuration, described below.
Error: Missing resource to override
│ 
│   on network_security_rule_override.tf line 23:
│   23: resource "azurerm_network_security_rule" "ssh-restricted" {
│ 
│ There is no azurerm_network_security_rule resource named "ssh-restricted".
│ An override file can only override a resource block defined in a primary
│ configuration file.
╵

╷
│ Error: Missing resource to override
│ 
│   on network_security_rule_override.tf line 39:
│   39: resource "azurerm_network_security_rule" "allow-https" {
│ 
│ There is no azurerm_network_security_rule resource named "allow-https". An
│ override file can only override a resource block defined in a primary
│ configuration file.
╵

╷
│ Error: Missing resource to override
│ 
│   on network_security_rule_override.tf line 52:
│   52: resource "azurerm_network_security_rule" "allow-strongbox" {
│ 
│ There is no azurerm_network_security_rule resource named "allow-strongbox".
│ An override file can only override a resource block defined in a primary
│ configuration file.
╵


```

**Comment by MarcPaquette** [comment]
Hi Folks,

Our team (CAPI) is also experiencing this issue.

Thanks,
Marc

**Comment by gbandres98** [comment]
Hi @giva01121 @MarcPaquette, can you please check if the v8.4.43 release fixes this issue? Thanks

**Comment by akop** [comment]
Hi @gbandres98, I already checked this and it seems the bbl v8.4.43 contains still the 0.15 terraform binary in the Mac OSX bbl build. 

I managed to work around the errors which @giva01121 faced with putting the mentioned non existing resources in a *-overwrite.tf. So it was like additions in -overwrite.tf and modifications in _overwrite.tf.
But this leads to other provider errors and at the end I gave up.

I think there are multiple steps necessary to migrate between the terraform versions.
So downgrading back to 0.11.15 is the easiest solution.

Not sure if there is the plan to upgrade the terraform version in future but then we need a migration path also if it means to spin up a new bbl env and migrate all the deployments.

Kind regards,
Andi
